src/psyclone/psyir/tools/extract_driver_creator.py

```diff
@@ -0,0 +1,560 @@
+    def create(self, nodes, input_list, output_list,
+               prefix, postfix, region_name):
+        # pylint: disable=too-many-arguments
+        '''This function uses the PSyIR to create a stand-alone driver
+        that reads in a previously created file with kernel input and
+        output information, and calls the kernels specified in the 'nodes'
+        PSyIR tree with the parameters from the file.
+
+        :param nodes: a list of nodes.
+        :type nodes: list of :py:obj:`psyclone.psyir.nodes.Node`
+        :param input_list: list of variables that are input parameters.
+        :type input_list: list of :py:class:`psyclone.core.Signature`
+        :param output_list: list of variables that are output parameters.
+        :type output_list: list or :py:class:`psyclone.core.Signature`
+        :param str prefix: the prefix to use for each PSyData symbol, \
+            e.g. 'extract' as prefix will create symbols `extract_psydata`.
+        :param str postfix: a postfix that is appended to an output variable \
+            to create the corresponding variable that stores the output \
+            value from the kernel data file. The caller must guarantee that \
+            no name clashes are created when adding the postfix to a variable \
+            and that the postfix is consistent between extract code and \
+            driver code (see 'ExtractTrans.determine_postfix()').
+        :param (str,str) region_name: an optional name to \
+            use for this PSyData area, provided as a 2-tuple containing a \
+            location name followed by a local name. The pair of strings \
+            should uniquely identify a region.
+
+        '''
+        # pylint: disable=too-many-locals
+
+        module_name, local_name = region_name
+
+        # module_name, region_name = self.region_identifier
+        unit_name = "{0}_{1}".format(module_name, local_name)
+
+        # First create the file container, which will only store the program:
+        file_container = FileContainer(unit_name)
+
+        # Create the program and add it to the file container:
+        program = Routine(unit_name, is_program=True)
+        program_symbol_table = program.symbol_table
+        file_container.addchild(program)
+
+        if prefix:
+            prefix = prefix + "_"
+
+        psy_data_mod = ContainerSymbol(prefix+"psy_data_mod")
+        program_symbol_table.add(psy_data_mod)
+        psy_data_type = DataTypeSymbol(prefix+"PsyDataType", DeferredType(),
+                                       interface=GlobalInterface(psy_data_mod))
+        program_symbol_table.add(psy_data_type)
+
+        writer = FortranWriter()
+        # The validation of the extract transform guarantees that all nodes
+        # in the node list have the same parent.
+        schedule_copy = nodes[0].parent.copy()
+        schedule_copy.lower_to_language_level()
+        ExtractDriverCreator.import_modules(program, schedule_copy)
```

Comment:
Yes, I know. In general I prefer that you can see in the caller that you are calling a static function (since you know this won't change the state of 'self'), but I've changed them.